### PR TITLE
[ASNIDE-17] Added AsnDocumentProcessor, AsnParsedDataStorage, AsnPars…

### PR DESCRIPTION
**Pull request by psockoN7**
…edDocument classes. Extracted AsnOutlineItemModel and renamed to AsnOverviewModel. Filled outline widget with better stubbs.

For now I decided it is better to finish the task 'as is' without adding planned AsnTypeAssignement class, since changes in this pull request are big and relevant enough. 

Now, the outline widget is filled with list of words present in file. Widget changes contents on document being edited. 

The flow of signals and objects possession differs in comparison to cpp as follows: 

_CppEditorWidget_ owns refernce to _CppEditorOutlineWidget_. _CppEditorOutlineWidget_ owns _OverviewModel_. On every new document update, new _CppEditorWidget_ is created. _CppEditor_ gets notifications on changes in document so the _CppEditorOutlineWidget_ updates _OverviewModel_. Update in _OverviewModel_ also triggers update in _OutlineModel_. This happens because _OverviewModel_ is shared between both _CppEditorOutlineWidget_ and _OutlineWidget_. _OutlineWidget_ on its creation receives refernce to _CppEditorWidget_, which has outline() function in its public API, so this way model can be shared among both outlines.

### CODE REVIEW

File: asndocument.cpp [hunk 1]
```diff
@@ -27,14 +27,26 @@
 
 #include "asnhighlighter.h"
 #include "asn1acnconstants.h"
+#include "asndocumentprocessor.h"
 
 using namespace Asn1Acn::Internal;
 
+const int PROCESS_DOCUMENT_INTERVAL_MS = 1000;
```
Comment (hcorg) [inline asndocument.cpp]: add `static` or anonymous namespace (although as long as it address is not taken, all consts are treated like "static", it's good to be sure ;) )

File: asndocumentprocessor.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+{
+}
+
+void AsnDocumentProcessor::run() const
+{
+    AsnParsedDataStorage *model = AsnParsedDataStorage::instance();
+
+    std::shared_ptr<AsnParsedDocument> doc = model->getDataForFile(m_filePath);
+    if (!doc || doc->getRevision() != m_textDocument->revision()) {
+        std::unique_ptr<AsnParsedDocument> asnFile(parse());
```
Comment (hcorg) [inline asndocumentprocessor.cpp]: imho `auto asnFile = parse();` could be more readable

File: asnparseddatastorage.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+using namespace Asn1Acn::Internal;
+
+static AsnParsedDataStorage *m_instance;
+
+AsnParsedDataStorage *AsnParsedDataStorage::instance()
+{
+    if (m_instance)
+        return m_instance;
+
+    m_instance = new AsnParsedDataStorage;
```
Comment (hcorg) [inline asnparseddatastorage.cpp]: can this method be called from multiple threads? I have no idea how many threads QtCreator uses internally - maybe we need to protect m_instance with some mutex guard.

Comment (psockoN7) [inline asnparseddatastorage.cpp]: I believe thread safeness issue with should now be resolved.

**Pull request merged**